### ISSUE TITLE
Option allowViewportOverflow seems to have no effect

### ISSUE THREAD

**Issue by utrotzek** [gemini-testing/hermione#478]
Hi guys, 

I'm trying to create screenshots of the current viewport like this:

```
.assertView(
    'viewport',
    '#my-selector',
    {
        ignoreElements: '#ignored-element',
        allowViewportOverflow: true,
        captureElementFromTop: false,
        compositeImage: false
    }
 );
```

Unfortunately the framework exists with the following error:

> Prepare screenshot failed with error type 'OUTSIDE_OF_VIEWPORT' and error message: Can not capture element, because it is outside of viewport. Try to set "captureElementFromTop=true" to scroll to it before capture.

I though the option **allowViewportOverflow** would prevent this. Is this not possible?

**Comment by DudaGod** [comment]
Hello.

> I though the option allowViewportOverflow would prevent this

Yes, it should be prevented if at least a part of your element `#my-selector` is inside current viewport. So you should set `captureElementFromTop: true` or scroll to it by yourself:
```
this.browser
  .execute(() => $('#my-selector').scrollIntoView())
  .assertView(...);
```

**Comment by utrotzek** [comment]
One part of `#my-selector`should be visible in the current viewport.

If I use the `captureElementFromTop: true`option it will take the screenshot without a problem. May it possible, that the top of `#my-selector` has to be in the current viewport? Because I'm quite sure, that in my test case at least something of the bottom and the middle of `#my-selector` is visible, but not the top.

Is there a way to take just a screenshot from the current viewport to investigate this?

**Comment by DudaGod** [comment]
Can you provide a code to reproduce your problem? We have tests on the logic that you described and everything works fine, capture only visible part of the element.

**Comment by utrotzek** [comment]
That's strange. I tried to provide a test code and in an independent test HTML I cannot reproduce this error. Obviously this has something to do with my project HTML/CSS. 
I'll investigate this.

**Comment by utrotzek** [comment]
I'm not able to narrow down the problem. As I said earlier, in my independent test HMTL everything works fine. In my real life project the error still occurs. I'm using the "body" selector to create the screenshot for now, which works in my use case.

In the meantime I had it also running in my real life project for a short time. I thought I had found the issue, it seemed that the test failed only if more than one test was executed in the same run. So I removed all other tests and executed only the one which was failing. Than it ran suddenly. After playing around a little bit more, the test fails now even if I execute only this test. So I'm at the beginning again.

BTW: I did not change anything in the project itself. Just played around with the test files.. 

I'm not sure how to debug this to find the issue. I can live with the "body" selector for now. But maybe someone has the same issue in the future.

Thanks for your help so far.
